spec/models/group_spec.rb

```diff
@@ -923,21 +923,35 @@ def can_view?(user, group)
   end
 
   describe '.search_groups' do
-    fab!(:group) { Fabricate(:group, name: 'tEsT_more_things', full_name: 'Abc something awesome') }
-    let(:messageable_group) { Fabricate(:group, name: "MessageableGroup", messageable_level: Group::ALIAS_LEVELS[:everyone]) }
+
+    def search_group_names(name)
+      Group.search_groups(name, sort: :auto).map(&:name)
+    end
 
     it 'should return the right groups' do
-      group
-
-      expect(Group.search_groups('te')).to eq([group])
-      expect(Group.search_groups('TE')).to eq([group])
-      expect(Group.search_groups('es')).to eq([group])
-      expect(Group.search_groups('ES')).to eq([group])
-      expect(Group.search_groups('ngs')).to eq([group])
-      expect(Group.search_groups('sOmEthi')).to eq([group])
-      expect(Group.search_groups('abc')).to eq([group])
-      expect(Group.search_groups('sOmEthi')).to eq([group])
-      expect(Group.search_groups('test2')).to eq([])
+      group_name = Fabricate(:group, name: 'tEsT_more_things', full_name: 'Abc something awesome').name
+
+      expect(search_group_names('te')).to eq([group_name])
+      expect(search_group_names('TE')).to eq([group_name])
+      expect(search_group_names('es')).to eq([group_name])
+      expect(search_group_names('ES')).to eq([group_name])
+      expect(search_group_names('ngs')).to eq([group_name])
+      expect(search_group_names('sOmEthi')).to eq([group_name])
+      expect(search_group_names('abc')).to eq([group_name])
+      expect(search_group_names('sOmEthi')).to eq([group_name])
+      expect(search_group_names('test2')).to eq([])
+    end
+
+    it 'should prioritize prefix matches' do
+      Fabricate(:group, name: 'pears_11', full_name: 'fred apple')
+      Fabricate(:group, name: 'apples', full_name: 'jane orange')
+      Fabricate(:group, name: 'oranges', full_name: 'ms fred')
+
+      expect(search_group_names('ap')).to eq(['apples', 'pears_11'])
+      expect(search_group_names('fr')).to eq(['pears_11', 'oranges'])
+      expect(search_group_names('oran')).to eq(['oranges', 'apples'])
```

Comment:
I think we'll also benefit from adding a test case where we have two prefix matches so that we can ensure stuff is ordered by `Group#name`

Reply:
sure I can add a test